onnxruntime/core/providers/rocm/roctracer_manager.h

```diff
@@ -0,0 +1,119 @@
+#pragma once
+
+#include <atomic>
+#include <map>
+#include <memory>
+#include <mutex>
+
+#include <hip/hip_runtime_api.h>
+#include <roctracer/roctracer.h>
+#include <roctracer/roctracer_hcc.h>
+#include <roctracer/roctracer_hip.h>
+#include <roctracer/roctracer_ext.h>
+#include <roctracer/roctracer_roctx.h>
+
+#include "core/common/profiler_common.h"
+#include "core/common/inlined_containers.h"
+
+namespace onnxruntime {
+namespace profiling {
+
+class RoctracerActivityBuffer {
+ public:
+  RoctracerActivityBuffer()
+      : data_(nullptr), size_(0) {}
+
+  RoctracerActivityBuffer(const char* data, size_t size)
+      : data_(std::make_unique<char[]>(size)), size_(size) {
+    memcpy(data_.get(), data, size);
+  }
+
+  RoctracerActivityBuffer(const RoctracerActivityBuffer& other)
+      : RoctracerActivityBuffer(other.data_.get(), other.size_) {}
+
+  RoctracerActivityBuffer(RoctracerActivityBuffer&& other)
+      : RoctracerActivityBuffer() {
+    std::swap(data_, other.data_);
+    std::swap(size_, other.size_);
+  }
+
+  RoctracerActivityBuffer& operator=(const RoctracerActivityBuffer& other);
+  RoctracerActivityBuffer& operator=(RoctracerActivityBuffer&& other);
+
+  // accessors
+  char* GetData() { return data_.get(); }
+  const char* GetData() const { return data_.get(); }
+  size_t GetSize() const { return size_; }
+
+ private:
+  std::unique_ptr<char[]> data_;
+  size_t size_;
+};
+
+struct ApiCallRecord {
+  uint32_t domain_;
+  uint32_t cid_;
+  hip_api_data_t api_data_{};
+};
+
+class RoctracerManager {
+ public:
+  RoctracerManager(const RoctracerManager&) = delete;
+  RoctracerManager& operator=(const RoctracerManager&) = delete;
+  RoctracerManager() = default;
```

Comment:
I think this should be `private` since this is a singleton.

Reply:
Good catch! Thank you!